tests/test_clarity.py

```diff
@@ -60,227 +58,249 @@ class FakeProcess2(FakeProcess):
     Mock(project=FakeEntity('that'), udf={'Species': 'a_species'})
 ]
 
+class TestClarity(TestEGCG):
+
+    def setUp(self):
+        clarity._lims = Mock()
+        clarity.app_logger = Mock()
+
+    def test_connection(self):
+        cfg.load_config_file(self.etc_config)
+        clarity._lims = None
+        # create connection from config
+        lims = clarity.connection()
+        assert lims.baseuri == 'a_baseuri/'
 
-def test_get_valid_lanes():
-    fake_flowcell = Mock(
-        placements={
-            '1:this': Mock(udf={'Lane Failed?': False}),
-            '2:that': Mock(udf={'Lane Failed?': False}),
-            '3:other': Mock(udf={'Lane Failed?': True})
-        }
-    )
-    with patched_lims('get_containers', [fake_flowcell]) as mocked_lims:
-        valid_lanes = clarity.get_valid_lanes('a_flowcell_name')
-        mocked_lims.assert_called_with(type='Patterned Flowcell', name='a_flowcell_name')
-        assert valid_lanes == [1, 2]
-
-
-def test_find_project_from_sample():
-    with patched_clarity('get_samples', fake_samples) as mocked_get_samples:
-        project_name = clarity.find_project_name_from_sample('a_sample')
-        mocked_get_samples.assert_called_with('a_sample')
-        assert project_name is None
-        clarity.app_logger.error.assert_called_with('%s projects found for sample %s', 2, 'a_sample')
-
-    with patched_clarity('get_samples', fake_samples[0:1]):
-        assert clarity.find_project_name_from_sample('a_sample') == 'this'
-
-
-@patched_lims('get_artifacts', [Mock(parent_process=Mock(udf={}, input_per_sample=lambda sample_name: [Mock(position='1:this', udf={})]))])
-@patched_clarity('get_sample', FakeEntity('a_sample'))
-def test_find_run_elements_from_sample(mocked_get_sample, mocked_get_artifacts):
-    assert list(clarity.find_run_elements_from_sample('a_sample')) == [(None, '1')]
-    mocked_get_sample.assert_called_with('a_sample')
-    mocked_get_artifacts.assert_called_with(sample_name='a_sample', process_type='AUTOMATED - Sequence')
-
+        # Check caching
+        assert lims == clarity.connection()
```

Comment:
If we're checking for object caching, this should be `lims is clarity.connection()`. We can probably trust that `clarity.connection() == clarity.connection()`!